test/testSOnxRnManifold.jl

```diff
@@ -0,0 +1,114 @@
+
+
+# Rigid Body Kinematics Metric CrokeKumar eq 61.
+# A family of left invariant metrics:
+# G = [αI 0; 0 βI]
+# where α and β are arbitrary constants, satisfies all the equations (80). 
+# This are the only left-invariant metrics which are compatible with the acceleration connection.
+# Can we add α and β as parameters to the metric?
+struct LeftInvariantRBKMetric <: RiemannianMetric end
+
+SOnxRnManifold(n) = MetricManifold(SpecialEuclideanGroup(n; variant=:right), LeftInvariantRBKMetric())
+
+SOnxRnManifoldType = Union{typeof(SOnxRnManifold(2)), typeof(SOnxRnManifold(3))}
+
+Manifolds.identity_element(::typeof(SOnxRnManifold(2))) = ArrayPartition(SA[0;0.0],SA[1 0; 0 1.0])
+Manifolds.identity_element(::typeof(SOnxRnManifold(3))) = ArrayPartition(SA[0,0,0.0],SA[1 0 0; 0 1 0; 0 0 1.0])
+
+# geodesics for metric (61) are the same as geodesics on the product manifold SO(3)×IR3
+function Manifolds.exp(M::SOnxRnManifoldType, p, X)
+    G = base_manifold(M)
+    ε = identity_element(M)
+    return compose(G, p, exp(base_manifold(G), ε, X))
+end
+
+function Manifolds.log(M::SOnxRnManifoldType, p, q)
+    G = base_manifold(M)
+    ε = identity_element(M)
+    X = log(base_manifold(G), ε, compose(G, inv(G, p), q))
+    return X
+end
+
+function Manifolds.inner(M::SOnxRnManifoldType, p, X, Y)
+    Xtr = submanifold_components(M, X)[1]
+    XRo = submanifold_components(M, X)[2]
+    Ytr = submanifold_components(M, Y)[1]
+    YRo = submanifold_components(M, Y)[2]
+    # Metric on Chirikjian, p35 W = diagm([1,1,2])
+    return dot(Xtr,Ytr) + dot(XRo, YRo)/2
+end
```

Comment:
@kellertuer thanks for the pointers to the MetricManifold, I played around with it here. I'm not sure I'm following the maths exactly right, but it seems to be doing what I think it should. Do you mind giving your thought on this before I proceed too far. 
I think the only difference between this and the default in Manifolds is the 2 scaling on the metric?

Reply:
Yes, that looks reasonable and fits the general idea we have how to implement these.

One thing that is maybe a bit misleading, is that your `identity_element` would always return static arrays even if someone works only with usual arrays. Is that intentional?
It would also always return a array partition even if someone would work with the (homogeneous) matrix representation).
We added a topical type parameter to that function (and to `zero_vector` on a Lie algebra) to make that more consistent.